avm/res/resources/deployment-script/main.json

```diff
@@ -492,4 +492,4 @@
       "value": "[split(reference('deploymentScriptLogs').log, '\n')]"
     }
   }
-}
\ No newline at end of file
+}
```

Comment:
Not sure what's happening here, must be the line ending just as with the `version.json` file. 

Just a note on that for awareness: If this PR goes in as is, the publishing step will run a `git diff` & detect that there was a change in both of these files. This will cause it to decide to publish a new version.
If they would not be part of the PR, the diff would only include the test files & readme, which cause the publishing script to ignore them and not publish a new version.

Long story short - I'd recommend to update the ending to their original form in both files 😏

Reply:
So what happens is `Set-AVM-Readme` does not create an empty line during the generation of the `main.json` to signalise EOF  but then `bicep build` creates an empty line which then overwrites the `main.json` during the `New-AzDeployment`.

![image](https://github.com/Azure/bicep-registry-modules/assets/19261257/eb8c25b5-6fa3-42e7-8073-5ca067fd27a0)

Sometimes when you test locally it might fail or get stuck leaving the `main.json` with an EOL resulting in this when one forgets to run `Set-AVMReadme` again to generate a `main.json` without an empty line.

Reply:
Hey @segraef, that's fair - and hopefully something we can somehow fix in the utility.
But for the time being, can you undo the change in the `version.json` file? If you don't the CI **will** publish a new version of the module - and given that the purpose of the PR was to add a test, I'm not sure that's your intention 😄

Reply:
Still needs to be undone 😉